xml/admin_ceph_upgrade.xml

```diff
@@ -203,340 +59,448 @@
       <para>
-       &igw;s
+       Changing a bucket type from <literal>straw</literal> to
+       <literal>straw2</literal> results in a small amount of data movement,
+       depending on how much the bucket item weights vary from each other. When
+       the weights are all the same, no data will move. When an item's weight
+       varies significantly there will be more movement. To migrate, execute:
       </para>
-     </listitem>
-     <listitem>
+<screen>&prompt.cephuser;ceph osd getcrushmap -o backup-crushmap
+&prompt.cephuser;ceph osd crush set-all-straw-buckets-to-straw2</screen>
       <para>
-       &ganesha;
+       If there are problems, you can revert this change with:
       </para>
-     </listitem>
-     <listitem>
+<screen>&prompt.cephuser;ceph osd setcrushmap -i backup-crushmap</screen>
       <para>
-       &sgw;s
+       Moving to <literal>straw2</literal> buckets unlocks a few recent
+       features, such as the <literal>crush-compat</literal> balancer mode that
+       was added in &ceph; Luminous (SES 4).
       </para>
-     </listitem>
-    </orderedlist>
-   </listitem>
-   <listitem>
-    <para>
-     If you used &aa; in either 'complain' or 'enforce' mode, you need to set a
-     &spillar; variable before upgrading. Because &cephos; ships with &aa; by
-     default, &aa; management was integrated into &deepsea; stage 0. The
-     default behavior in &productname; &productnumber; is to remove &aa; and
-     related profiles. If you want to retain the behavior configured in
-     &productname; &prevproductnumber;, verify that one of the following lines
-     is present in the <filename>/srv/pillar/ceph/stack/global.yml</filename>
-     file before starting the upgrade:
-    </para>
+     </important>
+    </listitem>
+    <listitem>
+     <para>
+      Check that &ceph; 'jewel' profile is used:
+     </para>
+<screen>
+&prompt.cephuser;ceph osd crush dump | grep profile
+</screen>
+    </listitem>
+   </itemizedlist>
+  </sect2>
+
+  <sect2 xml:id="before-upgrade-rbd-clients">
+   <title>Upgrade Old RBD Kernel Clients</title>
+   <para>
+    In case old RBD kernel clients (older than &prevcephos;) are being used,
+    refer to <xref linkend="rbd-old-clients-map" />. We recommend upgrading old
+    RBD kernel clients if possible.
+   </para>
+  </sect2>
+
+  <sect2 xml:id="before-upgrade-apparmor">
+   <title>Adjust &aa;</title>
+   <para>
+    If you used &aa; in either 'complain' or 'enforce' mode, you need to set a
+    &spillar; variable before upgrading. Because &cephos; ships with &aa; by
+    default, &aa; management was integrated into &deepsea; stage 0. The default
+    behavior in &productname; &productnumber; is to remove &aa; and related
+    profiles. If you want to retain the behavior configured in &productname;
+    &prevproductnumber;, verify that one of the following lines is present in
+    the <filename>/srv/pillar/ceph/stack/global.yml</filename> file before
+    starting the upgrade:
```

Comment:
Do we want the product-number to be a self-updating thing? It seems to me that a literal product number would be preferable here, but I could be wrong.